pkg/infrastructure/azure/dns.go

```diff
@@ -134,6 +133,19 @@ func createDNSEntries(ctx context.Context, in clusterapi.InfraReadyInput, extLBF
 	return nil
 }
 
+func isAPIServerPublic(in clusterapi.InfraReadyInput) bool {
+	switch in.InstallConfig.Config.Publish {
+	case types.ExternalPublishingStrategy:
+		return true
+	case types.InternalPublishingStrategy:
+		return false
+	case types.MixedPublishingStrategy:
+		return (in.InstallConfig.Config.OperatorPublishingStrategy.APIServer == "External")
+	default:
+		return true
+	}
+}
+
```

Comment:
nit: Probably can simplify to 
```suggestion
func isAPIServerPublic(in clusterapi.InfraReadyInput) bool {
	switch in.InstallConfig.Config.Publish {
	case types.InternalPublishingStrategy:
		return false
	case types.MixedPublishingStrategy:
		return (in.InstallConfig.Config.OperatorPublishingStrategy.APIServer == "External")
	}
	return true
}

```